### PR TITLE
feat: migrate from dep to mod

### PR DESCRIPTION
**Pull request by f0rmiga**
- Sub-moduled `pivotal-k8s/repo-infra` under `third-party/` and ignored it for the boilerplate verification.
- Removed the dep instructions from README since Go Modules are in the standard Go tooling.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	github.com/googleapis/gnostic v0.1.0 // indirect
+	github.com/gorilla/context v0.0.0-20160226214623-1ea25387ff6f // indirect
+	github.com/gorilla/mux v1.6.1 // indirect
+	github.com/gregjones/httpcache v0.0.0-20180305231024-9cad4c3443a7 // indirect
+	github.com/juju/ratelimit v1.0.1 // indirect
+	github.com/niemeyer/pretty v0.0.0-20200227124842-a10e7caefd8e // indirect
+	github.com/onsi/ginkgo v1.12.0 // indirect
+	github.com/onsi/gomega v1.10.0 // indirect
+	github.com/peterbourgon/diskv v2.0.1+incompatible // indirect
+	github.com/pkg/errors v0.8.1
```
Comment (mook-as) [inline go.mod:23]: v0.8.0 → v0.8.1 (which is probably okay).

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	github.com/pmorie/go-open-service-broker-client v0.0.0-20180304212357-e8aa16c90363
+	github.com/pmorie/osb-broker-lib v0.0.0-20180516212803-87d71cfbf342
+	github.com/prometheus/client_golang v1.6.0
```
Comment (mook-as) [inline go.mod:26]: v0.8.0 → v1.6.0 (I assume to pick up go modules compatibility or something?)

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	github.com/spf13/pflag v1.0.0 // indirect
+	github.com/stretchr/testify v1.5.1 // indirect
+	golang.org/x/crypto v0.0.0-20190424203555-c05e17bb3b2d // indirect
+	golang.org/x/text v0.3.2 // indirect
+	google.golang.org/genproto v0.0.0-20190502173448-54afdca5d873 // indirect
+	google.golang.org/grpc v1.20.1
```
Comment (mook-as) [inline go.mod:32]: v1.7.2 → v1.20.1 — that's a larger bump. I assume that's for go modules support?

File: go.mod [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	gopkg.in/check.v1 v1.0.0-20200227125254-8fa46927fb4f // indirect
+	gopkg.in/inf.v0 v0.9.1 // indirect
+	gopkg.in/yaml.v2 v2.2.5
```
Comment (mook-as) [inline go.mod:35]: `d670f9405373` (a random commit on the `v2` branch) → v2.2.5

File: go.mod (hunk 2 above)
Comment (f0rmiga) [reply go.mod:26]: Yes, this one I explicitly bumped to pick up the go modules compatibility.

File: go.mod (hunk 3 above)
Comment (f0rmiga) [reply go.mod:32]: This one I didn't bump manually, so Go might have figured that out.

File: .gitmodules [hunk 5]
```diff
@@ -0,0 +1,3 @@
+[submodule "third-party/repo-infra"]
+	path = third-party/repo-infra
+	url = https://github.com/pivotal-k8s/repo-infra.git
```
[the remark this answers is not in the document]
Comment (f0rmiga) [reply .gitmodules:3]: I simply ported from what we already had. I did read the source from 3 versions while doing this.

1. The [kubernetes/repo-infra](https://github.com/kubernetes/repo-infra/blob/fafebc939c8fb90e620ed348ec81fd4d05fe36bc/verify/verify-boilerplate.sh#L24) uses Bazel, and we don't (in this repository).
2. The tip of master of [pivotal-k8s/repo-infra](https://github.com/pivotal-k8s/repo-infra/blob/84d52408a061e87d45aebf5a0867246bdf66d180/verify/verify-boilerplate.sh#L26), overrides the `REPO_ROOT` env var.
3. The commit we picked on [pivotal-k8s/repo-infra](https://github.com/pivotal-k8s/repo-infra/blob/cefc8e98b61a8a3cd0d27564908e41439c8fe039/verify/verify-boilerplate.sh#L37) allows us to set `REPO_ROOT` env var, and it serves its purpose in our repository.

File: go.mod (hunk 4 above)
Comment (f0rmiga) [reply go.mod:35]: Also automatically done by Go mod.

File: .gitmodules (hunk 5 above)
[the remark this answers is not in the document]
Comment (mook-as) [reply .gitmodules:3]: I was going to say things, but then I clicked your links and realized the things I was about to say were dumb :D
Sounds good to me!

**Pull request merged**